conans/test/integration/options/test_options_build_requires.py

```diff
@@ -0,0 +1,96 @@
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+def test_build_requires_options_different():
+    # copied from https://github.com/conan-io/conan/pull/9839
+    client = TestClient()
+
+    conanfile_openssl_1_1_1 = GenConanfile("openssl", "1.1.1")
+    conanfile_openssl_3_0_0 = GenConanfile("openssl", "3.0.0") \
+        .with_option("no_fips", [True, False]) \
+        .with_default_option("no_fips", True)
+    conanfile_cmake = GenConanfile("cmake", "0.1") \
+        .with_requires("openssl/1.1.1")
+    conanfile_consumer = GenConanfile("consumer", "0.1") \
+        .with_build_requires("cmake/0.1") \
+        .with_requires("openssl/3.0.0")
+
+    client.save({"openssl_1_1_1.py": conanfile_openssl_1_1_1,
+                 "openssl_3_0_0.py": conanfile_openssl_3_0_0,
+                 "conanfile_cmake.py": conanfile_cmake,
+                 "conanfile.py": conanfile_consumer})
+
+    client.run("create openssl_1_1_1.py")
+    client.run("create openssl_3_0_0.py")
+    client.run("create conanfile_cmake.py")
+    client.run("install conanfile.py")
```

Comment:
so?

Reply:
Added checks and some explanations, as well as docstrings for tests